### PR TITLE
CB-2339: upate password policy

### PR DESCRIPTION
**Pull request by tiborpopovics**

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/PasswordPolicyService.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import com.sequenceiq.freeipa.client.model.PasswordPolicy;
+
+@Service
+class PasswordPolicyService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PasswordPolicyService.class);
+
+    private static final String MIN_PASSWORD_LENGTH = "krbpwdminlength";
+
+    private static final Integer MIN_PASSWORD_LENGTH_VALUE = 8;
```
Comment (keyki) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/PasswordPolicyService.java]: would be better if we make these values configurable @Value.. with the defaults

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/PasswordPolicyService.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    void updatePasswordPolicy(FreeIpaClient freeIpaClient) {
+        try {
+            PasswordPolicy config = freeIpaClient.getPasswordPolicy();
+            Map<String, Object> params = getParameters(config);
+            if (!params.isEmpty()) {
+                freeIpaClient.updatePasswordPolicy(params);
+            }
+        } catch (FreeIpaClientException e) {
+            LOGGER.error("Failed to create policy.", e);
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/PasswordPolicyService.java]: we should throw the error here to make freeipa provision fail

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/PasswordPolicyService.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    private static final String MAX_PASSWORD_LIFETIME_DAYS = "krbmaxpwdlife";
+
+    private static final String MAX_FAILURES_BEFORE_LOCK = "krbpwdmaxfailure";
+
+    private static final String FAILURE_RESET_INTERVALL_SECONDS = "krbpwdfailurecountinterval";
+
+    private static final String LOCKOUT_DURATION_SECONDS = "krbpwdlockoutduration";
+
+    @Value("passwordpolicy.min-password-lenght")
```
Comment (keyki) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/PasswordPolicyService.java]: does this work? Shouldn't it be freeipa.passwordpolicy..?

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/FreeIpaPasswordValidator.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.freeipa.controller.exception.BadRequestException;
+
+@Component
+class FreeIpaPasswordValidator {
+
+    private static final Set<String> PATTERNS = Set.of("(?=.*[a-z]).*$", "(?=.*[A-Z]).*$", "(?=.*[0-9]).*$", ".*\\W+.*");
```
Comment (abhishekaypurohit) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/FreeIpaPasswordValidator.java]: This should match with : https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/services/api/environments2/src/main/java/com/cloudera/thunderhead/service/environments2api/server/PasswordValidator.java#L16

Comment (keyki) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/FreeIpaPasswordValidator.java]: The policy is based on JIRA https://jira.cloudera.com/browse/CB-2339. Can you please update the jira so we can make the proper change here.

**Pull request merged**